src/quacc/calculators/espresso/espresso.py

```diff
@@ -101,11 +109,56 @@ def read_results(self, directory: Path | str) -> dict[str, Any]:
         dict
             The results dictionnary
         """
+
         results = read(Path(directory) / self.outputname, binary=self.binary)
         if "energy" not in results:
             results["energy"] = None
         return results
 
+    def _outdir_handler(self, parameters: dict[str, Any]) -> dict[str, Any]:
+        """
+        Function that handles the various outdir of espresso binaries.
+        If the user-supplied path are absolute, they are resolved and checked
+        against the scratch_dir and results_dir. If they are relative, they are
+        resolved against the scratch_dir and results_dir. If they are not in
+        either of those directories, they are ignored. This case is discouraged,
+        are encouraged to use quacc's scratch_dir and results_dir.
+
+        Parameters
+        ----------
+        parameters
+            User-supplied kwargs
+
+        Returns
+        -------
+        dict[str, Any]
+            The merged kwargs
+        """
+
+        input_data = parameters.get("input_data", {})
+
+        cwd = Path.cwd()
```

Comment:
I fear this may cause some confusion, and I generally recommend not relying on the `cwd` in quacc. Everything should be absolute whenever possible.

Let's say you want to run a two subsequent jobs. In quacc, what will happen is the following:

1. When `calc_setup()` is called, It'll `cd` into a custom directory to run Job 1 (the `tmpdir`).
2. The job will finish, and it will `cd` back into the directory where the Job 1 results are stored (`job_results_dir`). 
3. Job 2 will start, and it'll `cd` into a custom directory to run Job 2 _when `calc_setup` is called_.

But when the `Espresso` calculator is instantiated, you're still in the directory from mentioned in Step 2, which the user doesn't know about. There are also other, much more complicated concerns with using relative paths or `cwd` when running parallelized calculations with many workflow engines (notably parsl) that I won't bore you with.